pallets/parachain-staking/src/lib.rs

```diff
@@ -1745,26 +1744,29 @@ pub mod pallet {
 				}
 			};
 			let collator_fee = <CollatorCommission<T>>::get();
-			for (val, pts) in <AwardedPts<T>>::drain_prefix(round_to_payout) {
+			for (collator, pts) in <AwardedPts<T>>::drain_prefix(round_to_payout) {
 				let pct_due = Perbill::from_rational(pts, total);
 				let mut amt_due = pct_due * issuance;
 				// Take the snapshot of block author and nominations
-				let state = <AtStake<T>>::take(round_to_payout, &val);
+				let state = <AtStake<T>>::take(round_to_payout, &collator);
 				if state.nominators.is_empty() {
 					// solo collator with no nominators
-					mint(amt_due, val.clone());
+					mint(amt_due, collator.clone());
+					Self::deposit_event(Event::Rewarded(collator.clone(), collator.clone(), amt_due));
 				} else {
 					// pay collator first; commission + due_portion
 					let val_pct = Perbill::from_rational(state.bond, state.total);
 					let commission = collator_fee * amt_due;
 					amt_due -= commission;
 					let val_due = (val_pct * amt_due) + commission;
-					mint(val_due, val.clone());
+					mint(val_due, collator.clone());
+					Self::deposit_event(Event::Rewarded(collator.clone(), collator.clone(), val_due));
 					// pay nominators due portion
 					for Bond { owner, amount } in state.nominators {
 						let percent = Perbill::from_rational(amount, state.total);
 						let due = percent * amt_due;
-						increase_due_rewards(due, owner);
+						increase_due_rewards(due, owner.clone());
+						Self::deposit_event(Event::Rewarded(owner.clone(), collator.clone(), due));
```

Comment:
emits `Rewarded` event here but no transfer occurs, this is why I think a different event should just be used here

and as remarked in #806 , we don't really need the event to get this information so I'm not sure how necessary it is